macOS/DuckDuckGo/Preferences/Model/SyncPreferences.swift

```diff
@@ -543,6 +558,42 @@ extension SyncPreferences: ManagementDialogModelDelegate {
     }
 
     func startPollingForRecoveryKey(isRecovery: Bool) {
+        if featureFlagger.isFeatureOn(.exchangeKeysToSyncWithAnotherDevice) {
+            newStartPollingForRecoveryKey(isRecovery: isRecovery)
+        } else {
+            legacyStartPollingForRecoveryKey(isRecovery: isRecovery)
+        }
```

Comment:
Similarly here, let’s add a guard on the feature flag and call legacy implementation and return if not fulfilled, otherwise put the new implementation in place here.